CSharp/Library/Microsoft.Bot.Connector.Shared/CredentialProvider.cs

```diff
@@ -88,6 +88,11 @@ public static string GetAppSettings(string key)
         {
             return ConfigurationManager.AppSettings[key] ?? Environment.GetEnvironmentVariable(key, EnvironmentVariableTarget.Process);
         }
+
+        public static string NullIfEmpty(this string value)
```

Comment:
I don't think adding an Extension Method to "String" is necessarily a good idea. String is a very common class, and this behavior will be considered "Weird" by many.

Reply:
Sure. I'll remove this and add the check in line.